src/main/kotlin/org/openrewrite/kotlin/internal/KotlinParserVisitor.kt

```diff
@@ -1175,10 +1175,22 @@ class KotlinParserVisitor(
                 markers = markers.addIfAbsent(Extension(randomId()))
             }
 
-            val implicitExtensionFunction = functionCall is FirImplicitInvokeCall
+            var hasExplicitReceiver = false
+            if (functionCall is FirImplicitInvokeCall) {
+                val dispatchReceiver = functionCall.dispatchReceiver
```

Comment:
Typically, the compiler uses the explicit receiver if it's not null then uses the dispatch receiver otherwise.

Reply:
For the giving test case.
```
builder.init(TestInitializer("null"))
```
it has both an explicit receiver and a dispatch receiver.
however, the explicit receiver is `init` but not `builder`. but the dispatch receiver is the `builder`.

Reply:
I checked line `1180`, and the values are equal. The explicit receiver and the dispatch receiver are the same object. The compiler typically uses the explicit receiver when it is not null and falls back to the dispatch receiver otherwise. The code only uses the dispatch receiver, so I'm trying to understand what you mean.

Reply:
Yeah, the explicit receiver and dispatch receiver are equal here, revised to use the explicit receiver.